pages/api/user/[id].ts

```diff
@@ -25,8 +26,28 @@ export default async function handle(
         }
         ResponseUtil.returnOK(res, user);
         return;
+    } else if (req.method == "PUT") {
+        const session = await getSession({ req });
+        const userId = session ? session.id : req.query.id;
```

Comment:
Let's also store the user's role in nextauth callback so we can retrieve it from the `session`.

Reply:
Should this be in this PR or a separate one?

Reply:
I can make the change.